pkg/manager/member/pump_member_manager.go

```diff
@@ -0,0 +1,475 @@
+			klog.Warningf("hash collision detected on configmap: %s, update configmap content in-place", newCm.Name)
+		}
+		cm := *oldCm
+		cm.Data = newCm.Data
+		if isOrphan {
+			cm.OwnerReferences = newCm.OwnerReferences
+			cm.Labels = newCm.Labels
+		}
+		_, err = pmm.cmControl.UpdateConfigMap(tc, newCm)
+		if err != nil {
+			return nil, err
+		}
+		return newCm, err
+	}
+
+	return oldCm, nil
+}
+
+func getNewPumpHeadlessService(tc *v1alpha1.TidbCluster) *corev1.Service {
+	if tc.Spec.Pump == nil {
+		return nil
+	}
+
+	objMeta, pumpLabel := getPumpMeta(tc, controller.PumpPeerMemberName)
+
+	return &corev1.Service{
+		ObjectMeta: objMeta,
+		Spec: corev1.ServiceSpec{
+			ClusterIP: "None",
+			Ports: []corev1.ServicePort{
+				{
+					Name:       "pump",
+					Port:       8250,
+					TargetPort: intstr.FromInt(8250),
+					Protocol:   corev1.ProtocolTCP,
+				},
+			},
+			Selector:                 pumpLabel,
+			PublishNotReadyAddresses: true,
+		},
+	}
+}
+
+// getNewPumpConfigMap returns a configMap for pump
+func getNewPumpConfigMap(tc *v1alpha1.TidbCluster) (*corev1.ConfigMap, error) {
+	spec := tc.Spec.Pump
+	objMeta, _ := getPumpMeta(tc, controller.PumpMemberName)
+
+	buff := new(bytes.Buffer)
+	encoder := toml.NewEncoder(buff)
+	err := encoder.Encode(spec.Config)
+	if err != nil {
+		return nil, err
+	}
+	data := buff.Bytes()
+
+	name := controller.PumpMemberName(tc.Name)
+	if spec.ConfigUpdateStrategy == v1alpha1.ConfigUpdateStrategyRollingUpdate {
+		sum := sha256.Sum256(data)
+		suffix := fmt.Sprintf("%x", sum)[0:7]
+		name = fmt.Sprintf("%s-%s", name, suffix)
+	}
+	objMeta.Name = name
+
+	return &corev1.ConfigMap{
+		ObjectMeta: objMeta,
+		Data: map[string]string{
+			"pump-config": string(data),
+		},
+	}, nil
+}
+
+func getNewPumpStatefulSet(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap) (*appsv1.StatefulSet, error) {
+	spec, ok := tc.BasePumpSpec()
+	if !ok {
+		return nil, nil
+	}
+	objMeta, pumpLabel := getPumpMeta(tc, controller.PumpMemberName)
+	replicas := tc.Spec.Pump.Replicas
+	storageClass := tc.Spec.Pump.StorageClassName
+	podAnnos := CombineAnnotations(controller.AnnProm(8250), spec.Annotations())
+	storageRequest, err := controller.ParseStorageRequest(tc.Spec.Pump.Requests)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse storage request for pump, tidbcluster %s/%s, error: %v", tc.Namespace, tc.Name, err)
+	}
+	startScript, err := getPumpStartScript(tc)
+	if err != nil {
+		return nil, fmt.Errorf("cannot render start-script for pump, tidbcluster %s/%s, error: %v", tc.Namespace, tc.Name, err)
+	}
+
+	var envs []corev1.EnvVar
+	if tc.Spec.Pump.SetTimeZone != nil && *tc.Spec.Pump.SetTimeZone {
+		envs = append(envs, corev1.EnvVar{
+			Name:  "TZ",
+			Value: tc.Spec.Timezone,
+		})
+	}
+	if spec.HostNetwork() {
+		// For backward compatibility, set HOSTNAME to POD_NAME in hostNetwork mode
+		envs = append(envs, corev1.EnvVar{
```

Comment:
This can avoid rolling-update when user upgrade from helm chart, but I'm not sure if there is any side-effect, @cofyc PTAL

Reply:
you don't do this because this is a new sts to deploy (no need to keep back-compatibility like pd/tikv sts) 

you can always use `POD_NAME` for both pod network and host network mode because sts member address is always `<pod-name>-<pod-ordinal>`.

Reply:
See the comment below, we have to consider the helm chart.

Reply:
Actually I prefer to use `POD_NAME`, which is more accurate and has no hidden tricks, but we have to document that adopt pump via `TidbCluster` CR will trigger an rolling-update, this a trade-off, either looks not good but acceptable for me.

/cc @weekface @tennix how do you think?

Reply:
What about adding a feature gate for adopting helm managed resources?

Reply:
No gate is needed, if user leave the `.spec.pump` as nil, the helm resource will not be adopted. The real concern is that whether user has the ability to **adopt pump cluster and do not trigger rolling-update**. If not, abandon helm for existing cluster will require extra effort.

Reply:
adapt old pump clusters and do not trigger rolling-update when update tidb-operator is reasonable for the user.

Reply:
adapt old pump clusters and do not trigger rolling-update when update tidb-operator is reasonable for the user.

Reply:
So users can allow the adoption when they need to update pump clusters? If so, then it's reasonable and we can document about this.

Reply:
OK, if both approaches make sense, I pretend to keep it as is to avoid additional code change (mainly in e2e, there is a case to verify the sts do not get rolling-update after adoption, which need to be re-written if we choose to change the podTemplate in this PR), further improvement on the `PodTemplate` of pump could be made through other enhancement issues.